api-server/scanner-lib/src/blockchain_state/mod.rs

```diff
@@ -30,12 +30,18 @@ pub enum BlockchainStateError {
 }
 
 pub struct BlockchainState<S: ApiServerStorage> {
-    storage: S,
+    storage: Option<S>,
```

Comment:
I fixed this by creating an async clone.

Reply:
> So in your tests, you could just do local_node.storage().clone()

Once the storage was passed into the state, I needed it to be taken out again but couldn't because of a RwLock... but your change looks like it fixes my hack around the issue. Nice!